lib/broccoli/ember-app.js

```diff
@@ -467,6 +464,22 @@ class EmberApp {
     }
   }
 
+  _addJqueryInLegacyEmber() {
+    let emberJQuery = this.project.findAddonByName('@ember/jquery');
+    if (emberJQuery) {
+      this.vendorFiles['jquery.js'] = emberJQuery.paths.jquery;
+    } else {
+      let ember = this.project.findAddonByName('ember-source');
+      if (ember) {
+        if (semver.lt(this.project.addonPackages['ember-source'].pkg.version, '3.3.0')) {
```

Comment:
We should add some additional logic here:

1. if `config/optional-features.json` is present **AND** the (still to be created) `jquery-integration` optional feature flag is set to `true` or is not defined: we should continue to use `ember.paths.jquery` and issue a console warning/deprecation
2. if `config/optional-features.json` is _not_ present we issue a console warning/deprecation suggesting opting out of the feature via `@ember/optional-features` **or** adding the `@ember/jquery` addon